build-logic/convention/src/main/kotlin/AndroidApplicationConventionPlugin.kt

```diff
@@ -0,0 +1,122 @@
+            apply(plugin = libs.plugins.ksp.getPluginId())
+            apply(plugin = libs.plugins.hilt.getPluginId())
+            apply(plugin = libs.plugins.kotlin.parcelize.getPluginId())
+            apply(plugin = libs.plugins.compose.compiler.getPluginId())
+
+            extensions.configure<ApplicationExtension> {
+                namespace = APPLICATION_ID
+                compileSdk = libs.versions.androidSdk.compile.get().toInt()
+
+                defaultConfig {
+                    applicationId = APPLICATION_ID
+                    minSdk = libs.versions.androidSdk.min.get().toInt()
+                    targetSdk = libs.versions.androidSdk.target.get().toInt()
+
+                    versionName = project.version.toString()
+                    versionCode = System.getenv("VERSION_CODE")?.toIntOrNull() ?: 1
+                }
+
+                buildFeatures {
+                    viewBinding = true
+                    compose = true
+                    buildConfig = true
+                }
+
+                compileOptions {
+                    isCoreLibraryDesugaringEnabled = true
+                    sourceCompatibility(libs.versions.javaVersion.get())
+                    targetCompatibility(libs.versions.javaVersion.get())
+                }
+
+                tasks.withType<KotlinCompile>().configureEach {
+                    compilerOptions {
+                        jvmTarget.set(JvmTarget.fromTarget(libs.versions.javaVersion.get()))
+                    }
+                }
+
+                signingConfigs {
+                    create("release") {
+                        storeFile = file(System.getenv("KEYSTORE_PATH") ?: "release_keystore.keystore")
+                        storePassword = System.getenv("KEYSTORE_PASSWORD") ?: ""
+                        keyAlias = System.getenv("KEYSTORE_ALIAS") ?: ""
+                        keyPassword = System.getenv("KEYSTORE_ALIAS_PASSWORD") ?: ""
+                        enableV1Signing = true
+                        enableV2Signing = true
+                    }
+                }
+
+                buildTypes {
+                    named("debug").configure {
+                        applicationIdSuffix = ".debug"
+                    }
+                    named("release").configure {
+                        isDebuggable = false
+                        isJniDebuggable = false
+                        signingConfig = signingConfigs.getByName("release")
+                    }
+                }
+
+                flavorDimensions.add("version")
+                productFlavors {
+                    create("minimal") {
+                        applicationIdSuffix = ".minimal"
+                        versionNameSuffix = "-minimal"
+                    }
+                    create("full") {
+                        applicationIdSuffix = ""
+                        versionNameSuffix = "-full"
+                    }
```

Comment:
Is it an issue for the watch to have these flavors?

Reply:
You're wasting some resources building it, and there is no such this as a GMS-less Wear OS so no reason for a (minimal) flavor to exist.

Reply:
If we go further with this PR, I'll extract this into a dedicated convention plugin that is not applied to Wear module.

Reply:
@jpelgrom I did extract this into another convention plugin to make it cleaner.